x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,518 @@
+        # password: "secret"
+        # If kube_config is not set, KUBECONFIG environment variable will be checked
+        # and if not present it will fall back to InCluster
+        # kube_config: ~/.kube/config
+        streams:
+          - id?: {id}
+            metricset: node
+            dataset: kubernetes.node
+            period: 10s
+          - id?: {id}
+            metricset: system
+            dataset: kubernetes.system
+            period: 10s
+          - id?: {id}
+            metricset: pod
+            dataset: kubernetes.pod
+            period: 10s
+          - id?: {id}
+            metricset: container
+            dataset: kubernetes.container
+            period: 10s
+          - id?: {id}
+            metricset: container
+            dataset: kubernetes.container
+            period: 10s
+          - id?: {id}
+            metricset: event
+            dataset: kubernetes.event
+            period: 10s
+      - type: metrics/kubernetes
+        hosts: ["kube-state-metrics:8080"]
+        streams:
+          - id?: {id}
+            metricset: state_node
+            dataset: kubernetes.state_node
+            period: 10s
+          - id?: {id}
+            metricset: state_deployment
+            dataset: kubernetes.state_deployment
+            period: 10s
+          - id?: {id}
+            metricset: state_replicaset
+            dataset: kubernetes.state_replicaset
+            period: 10s
+          - id?: {id}
+            metricset: state_statefulset
+            dataset: kubernetes.state_statefulset
+            period: 10s
+          - id?: {id}
+            metricset: state_pod
+            dataset: kubernetes.state_pod
+            period: 10s
+          - id?: {id}
+            metricset: state_container
+            dataset: kubernetes.state_container
+            period: 10s
+          - id?: {id}
+            metricset: state_container
+            dataset: kubernetes.state_container
+            period: 10s
+          - id?: {id}
+            metricset: state_cronjob
+            dataset: kubernetes.state_cronjob
+            period: 10s
+          - id?: {id}
+            metricset: state_resourcequota
+            dataset: kubernetes.state_resourcequota
+            period: 10s
+          - id?: {id}
+            metricset: state_service
+            dataset: kubernetes.state_serice
+            period: 10s
+          - id?: {id}
+            metricset: state_persistentvolume
+            dataset: kubernetes.state_persistentvolume
+            period: 10s
+          - id?: {id}
+            metricset: state_persistentvolumeclaim
+            dataset: kubernetes.state_persistentvolumeclaim
+            period: 10s
+
+  #################################################################################################
+  # - Kubernetes
+  - id?: my-docker
+    title: Collect docker
+    package:
+      name: epm/docker
+      version: 1.7.0
+    inputs:
+      - type: metrics/docker
```

Comment:
Nit: I keep stumbling over the type. I would expect it to be `docker/metrics`. The reason is the order I think of collecting: Let me get some data from Docker and select only the metrics. I understand that from an agent perspective it is the other way around, select metricbeat and then run docker.

Reply:
I don't have a strong feeling about metric before of after docker, I think your reasoning makes sense and might be more future proof if we do want to collect other thing. The mnemonic might be better.

Reply:
I've made the change on the config.